pkg/internal/validation/dns.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// This file's code was modified from "k8s.io/apimachinery/pkg/util/validation"
+// to avoid package dependencies. In case of additional functionality from
+// "k8s.io/apimachinery" is needed, re-consider whether to add the dependency.
+
+const (
+	dns1123LabelFmt     string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+	dns1123SubdomainFmt string = dns1123LabelFmt + "(\\." + dns1123LabelFmt + ")*"
+	dns1035LabelFmt     string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+)
+
+type dnsValidationConfig struct {
```

Comment:
@Adirio @camilamacedo86 refactored these for convenience after rebasing onto the feature branch, PTAL.

Reply:
I do like the refactor you did, avoiding code duplications through the use of the `dnsValidationConfig` struct and its `check` method. A couple of considerations:

- When I read `IsDNS1XXXLabel` I instantly imagine a `func (string) bool` signature. As the different errors need to be differentiate, we can not change the function signature to that one, so what about changing the name? Maybe around `MustBe...` or `ConformsTo...`. I guess we could check how the standard library 
- Returning a `[]string` containing the errors doesn't seem very idiomatic in Go. Maybe we should just return an `error` and have the user solve the problem in two steps in case he encounters two errors, as the second one will not be reported until the first one is solved.
- Would it make sense to group functions for the same specification? Something in the line of `validation.DNS1XXX.ConformsToLabel(myString)`. You would have either to export the struct instance or an interface to it:

```Golang
type DNSValidator interface {
	ConformsToLabel(string) error
	ConformsToSubdomain(string) error
}

var (
	DNS1023 DNSValidator = dns1023Config
	DNS1135 DNSValidator = dns1135Config
)
```

Reply:
The naming and return values could use some work. Doing this in a follow-up sounds good to me.